index.js

```diff
@@ -600,6 +616,17 @@ class Stream extends EventEmitter {
 
     return super.on(name, fn)
   }
+
+  static drained (stream) {
+    if (stream.destroyed) return Promise.resolve(false)
```

Comment:
I don't really follow the code but just in case, `destroyed` vs `destroying`?

Reply:
destroyed here is fine, cause these are resolved in the final teardown handler, normally otherwise yes.